inference-engine/src/vpu/graph_transformer/src/middleend/passes/propagate_dynamism_to_outputs.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/middleend/pass_manager.hpp"
+
+#include <set>
+#include <memory>
+
+namespace vpu {
+
+namespace {
+
+class PassImpl final : public Pass {
+public:
+    explicit PassImpl(StageBuilder::Ptr stageBuilder) : _stageBuilder(std::move(stageBuilder)) {}
+
+    void run(const Model& model) override {
+        static const std::unordered_set<StageType> types = {StageType::Convert};
+
+        for (auto const& data : model->datas()) {
+            if (data->usage() != DataUsage::Output || data->parentDataToShapeEdge() != nullptr) {
+                continue;
+            }
+
+            auto const& producer = data->producer();
+            VPU_THROW_UNLESS(producer, "Output data must have a producer, but {} doesn't have", data->name());
+
+            if (types.count(producer->type()) == 0) {
+                continue;
+            }
+
+            VPU_THROW_UNLESS(producer->numInputs() == 1,
+                "Only single input producers are supported, but {} has {} inputs",
+                producer->name(), producer->numInputs());
+
+            auto const& input = producer->input(0);
+            auto const& parentDataToShapeEdge = input->parentDataToShapeEdge();
+            if (parentDataToShapeEdge == nullptr) {
+                continue;
+            }
+            auto const parent = parentDataToShapeEdge->parent();
+
+            // Create the second output with shape in case of dynamic output
+            const auto& shapeOutput = model->addOutputData(parent->name() + "@output-dynamic-shape", parent->desc());
```

Comment:
Output shape should have `@shape` suffix, isn't it?

Reply:
Nice catch, thanks! Fixed.